### PR TITLE
[Port from snakeyaml-engine] Issue 54: Insert a trailing space when alias is a simple key

### PR DESCRIPTION
**Pull request by krzema12**
Closes https://github.com/krzema12/snakeyaml-engine-kmp/issues/294

Ports:
* https://bitbucket.org/snakeyaml/snakeyaml-engine/commits/1e74afe280795e6480ec72e7efee5afed44f2e25
* https://bitbucket.org/snakeyaml/snakeyaml-engine/commits/089376b4d2bbd7a0ee66808b7c1be8acc99feb04

### CODE REVIEW

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import io.kotest.matchers.string.shouldContain
+import it.krzeminski.snakeyaml.engine.kmp.api.Dump
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+
+/**
+ * Issue 54: add a space after anchor (when it is a simple key)
+ */
+class DumpWithoutSpaceTest : FunSpec({
```
Comment (krzema12) [inline src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt:16]: TODO: port to common sources.

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    test("The output does include a space after the *1 alias") {
+        val obj = parse(
+            """|--- &1
+               |hash:
+               |  :one: true
+               |  :two: true
+               |  *1 : true""".trimMargin()
+        )
+        obj shouldNotBe null
```
Comment (krzema12) [inline src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt:46]: There's some kind of an infinite loop here which makes the test time out. Digging deeper.

Comment (krzema12) [reply src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt]: Commenting out this assertion makes the tests pass. Let's investigate why the assertion fails, and if obj is really null here.

Comment (krzema12) [reply src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt]: It turned out it's enough to replace it with `obj.shouldNotBeNull()`.

**Review by krzema12 — COMMENTED** [review]
# :warning: **Performance Alert** :warning:

Possible performance regression was detected for benchmark **'SnakeKMP benchmarks'**.
Benchmark result of this commit is worse than the previous benchmark result exceeding threshold `1.50`.

| Benchmark suite | Current: 7c1f7e8ee23dc520eec3462e309f0b004a5695e2 | Previous: 6bdb3c5036f521f92f438bdfc9ad89d804ea245f | Ratio |
|-|-|-|-|
| `macosX64.LoadingTimeBenchmark.loadsOpenAiSchema ( {"openAiYamlPath":"data/issues/kmp-issue-204-OpenAI-API.yaml"} )` | `226.50486293333333` ms/op | `118.67058208409091` ms/op | `1.91` |

This comment was automatically generated by [workflow](https://github.com/krzema12/snakeyaml-engine-kmp/actions?query=workflow%3ARun%20Benchmarks) using [github-action-benchmark](https://github.com/marketplace/actions/continuous-benchmark).

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                """|--- &1
+                   |hash:
+                   |  :one: true
+                   |  :two: true
+                   |  *1: true""".trimMargin()
+            )
+            fail("Shouldn't reach here!")
+        } catch (e: Exception) {
+            e.message shouldContain "could not find expected ':'"
+        }
```
Comment (OptimumCode) [inline src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt:36]: suggestion: you can replace try-catch using `shouldThrow<Exception> {}` function from kotest

**Review by OptimumCode — DISMISSED** [review]

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt (hunk 3 above)
Comment (krzema12) [reply src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt]: Right, here: https://github.com/krzema12/snakeyaml-engine-kmp/pull/309/commits/1faf96c932e272abc1b6d820cf004b322ade77e6

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**